test/append.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var eq = require('./utils').eq;
+var S = require('..');
+
+
+describe('append', function() {
+
+  it('adds the element to the end of the list', function() {
```

Comment:
Currently the `[a, $.Array(a), $.Array(a)]` portion of the definition is untested. Let's add this:

```diff
 describe('append', function() {

+  it('is a binary function', function() {
+    eq(typeof S.append, 'function');
+    eq(S.append.length, 2);
+  });
+
+  it('type checks its arguments', function() {
+    throws(function() { S.append('c', 'ab'); },
+           errorEq(TypeError,
+                   'Invalid value\n' +
+                   '\n' +
+                   'append :: a -> Array a -> Array a\n' +
+                   '               ^^^^^^^\n' +
+                   '                  1\n' +
+                   '\n' +
+                   '1)  "ab" :: String\n' +
+                   '\n' +
+                   'The value at position 1 is not a member of ‘Array a’.\n'));
+
+    throws(function() { S.append('c', [1, 2]); },
+           errorEq(TypeError,
+                   'Type-variable constraint violation\n' +
+                   '\n' +
+                   'append :: a -> Array a -> Array a\n' +
+                   '          ^          ^\n' +
+                   '          1          2\n' +
+                   '\n' +
+                   '1)  "c" :: String\n' +
+                   '\n' +
+                   '2)  1 :: Number, FiniteNumber, NonZeroFiniteNumber, Integer, ValidNumber\n' +
+                   '    2 :: Number, FiniteNumber, NonZeroFiniteNumber, Integer, ValidNumber\n' +
+                   '\n' +
+                   'Since there is no type of which all the above values are members, the type-variable constraint has been violated.\n'));
+  });
+
   it('adds the element to the end of the list', function() {
     eq(S.append('c', ['a', 'b']), ['a', 'b', 'c']);
```